metashare/api/jobs.py

```diff
@@ -0,0 +1,168 @@
+import contextlib
+import logging
+import traceback
+
+from asgiref.sync import async_to_sync
+from channels.layers import get_channel_layer
+from cumulusci.core.config import BaseProjectConfig, ScratchOrgConfig
+from cumulusci.core.runtime import BaseCumulusCI
+from django.utils.text import slugify
+from django.utils.timezone import now
+from django_rq import job
+from github3 import login
+from github3.exceptions import UnprocessableEntity
+
+from .github_context import (
+    extract_owner_and_repo,
+    get_cumulus_prefix,
+    local_github_checkout,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def report_scratch_org_error(instance, message):
+    from .serializers import ScratchOrgSerializer
+
+    message = {
+        "type": "SCRATCH_ORG_PROVISION_FAILED",
+        "payload": {"message": message, "model": ScratchOrgSerializer(instance).data},
+    }
+    model_name = instance._meta.model_name
+    id = str(instance.id)
+    group_name = "{model}.{id}".format(model=model_name, id=id)
+    channel_layer = get_channel_layer()
+    sent_message = {"type": "notify", "group": group_name, "content": message}
+    async_to_sync(channel_layer.group_send(group_name, sent_message))
```

Comment:
@wlonk I hit an error creating a scratch org, but it was not exposed at all. Console:

```
/app/metashare/api/jobs.py:36: RuntimeWarning: coroutine 'RedisChannelLayer.group_send' was never awaited
  async_to_sync(channel_layer.group_send(group_name, sent_message))
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

```
Traceback (most recent call last):
  File "/app/metashare/api/jobs.py", line 42, in report_errors_on
    yield
  File "/app/metashare/api/jobs.py", line 86, in create_scratch_org
    org.create_org()
  File "/usr/local/lib/python3.7/site-packages/cumulusci/core/config/ScratchOrgConfig.py", line 206, in create_org
    raise ScratchOrgException(message)
cumulusci.core.exceptions.ScratchOrgException: Failed to create scratch org: 

ERROR running force:org:create:  The signup request failed because this organization has reached its active scratch org limit

id=no-job-id at=ERROR time="2019-09-04 14:04:52.361339" tag=external module=worker msg="cumulusci.core.exceptions.ScratchOrgException: Failed to create scratch org: 

ERROR running force:org:create:  The signup request failed because this organization has reached its active scratch org limit
Traceback (most recent call last):
  File \"/usr/local/lib/python3.7/site-packages/rq/worker.py\", line 822, in perform_job
    rv = job.perform()
  File \"/app/metashare/rq_worker.py\", line 12, in wrapper
    return orig(self)
  File \"/usr/local/lib/python3.7/site-packages/rq/job.py\", line 605, in perform
    self._result = self._execute()
  File \"/usr/local/lib/python3.7/site-packages/rq/job.py\", line 611, in _execute
    return self.func(*self.args, **self.kwargs)
  File \"/app/metashare/api/jobs.py\", line 167, in create_branches_on_github_then_create_scratch_org_job
    scratch_org=scratch_org, user=user, repo_url=repo_url, commit_ish=commit_ish
  File \"/app/metashare/api/jobs.py\", line 86, in create_scratch_org
    org.create_org()
  File \"/usr/local/lib/python3.7/site-packages/cumulusci/core/config/ScratchOrgConfig.py\", line 206, in create_org
    raise ScratchOrgException(message)
cumulusci.core.exceptions.ScratchOrgException: Failed to create scratch org: 

ERROR running force:org:create:  The signup request failed because this organization has reached its active scratch org limit
"
```

Reply:
I'm getting that error over the websocket now. Can you confirm?